### PR TITLE
fix deprecation warnings

### PR DESCRIPTION
**Pull request by jakemac53**
Migration from `whereNotNull()` to `nonNulls`.

Should fix the analysis failures at HEAD.

### CODE REVIEW

File: build_runner_core/lib/src/changes/build_script_updates.dart [hunk 1]
```diff
@@ -51,7 +51,7 @@ class _MirrorBuildScriptUpdates implements BuildScriptUpdates {
     try {
       allSources = _urisForThisScript
           .map((id) => idForUri(id, packageGraph))
-          .whereNotNull()
+          .nonNulls
```
Comment (kevmoo) [inline build_runner_core/lib/src/changes/build_script_updates.dart:54]: This was IN the SDK before 3.5.0-...

Why bump the constraint?

Comment (jakemac53) [reply build_runner_core/lib/src/changes/build_script_updates.dart:54]: Analyzer

Comment (jakemac53) [reply build_runner_core/lib/src/changes/build_script_updates.dart:54]: (this bump should have actually landed in https://github.com/dart-lang/build/pull/3719)

Comment (kevmoo) [reply build_runner_core/lib/src/changes/build_script_updates.dart:54]: The SDK min bump is required by analyzer?

**Review by natebosch — APPROVED** [review]

**Pull request merged**